lib/rummager.rb

```diff
@@ -105,11 +99,18 @@
 require "govuk_index/presenters/parts_presenter"
 require "govuk_index/presenters/specialist_presenter"
 require "govuk_index/publishing_event_processor"
-require "govuk_index/publishing_event_worker"
 require "govuk_index/supertype_updater"
-require "govuk_index/supertype_worker"
 require "govuk_message_queue_consumer"
 
+require "sidekiq/base_job"
+require "sidekiq/govuk_index/page_traffic_job"
+require "sidekiq/govuk_index/popularity_job"
+require "sidekiq/govuk_index/publishing_event_job"
+require "sidekiq/govuk_index/supertype_job"
+require "sidekiq/indexer/amend_job"
+require "sidekiq/indexer/bulk_index_job"
+require "sidekiq/indexer/delete_job"
```

Comment:
Sorry I missed this last time. I think we risk increasing confusion of navigating this codebase by putting things in a sidekiq directory and then using the existing modular namespaces. For example if you want to find the class GovukIndex::PageTrafficJob you look in sidekiq/govuk_index otherwise a different class in that namespace is over in lib/govuk_index.

I think we're a bit caught between a Rails recommendation (app/workers => app/sidekiq) meeting a `lib` where the directories refer to namespacing.

I think the way to resolve this that causes the least pain would be to do the following:

- move `lib/sidekiq/base_job.rb` to `lib/base_job.rb`
- move sidekiq jobs in `lib/sidekiq/govuk_index` to `lib/govuk_index`
- move sidekiq jobs in `lib/sidekiq/indexer` to `lib/indexer`

---

An alternative would be to create a namespace for jobs and put them all in that, but that seems much more fiddly trying to get the names of these things appropriate without the qualifying namespace. Does feel a bit janky having the jobs mixed into lib code.

Reply:
Thanks for the review @kevindew 

I appreciate that moving the jobs outside of that lib dir does make it slightly harder to parse together what's going on.

Having said that all other codebases on GOV.UK will end up following the pattern of adding Sidekiq based on the guidance provided [here](https://github.com/alphagov/govuk_sidekiq/blob/main/CHANGELOG.md#900) by Bruce though. 

Do you think it would feel slightly off to have this be the only repo that doesn't?

I guess these 

```
require "sidekiq/govuk_index/page_traffic_job"
require "sidekiq/govuk_index/popularity_job"
require "sidekiq/govuk_index/publishing_event_job"
require "sidekiq/govuk_index/supertype_job"
require "sidekiq/indexer/amend_job"
require "sidekiq/indexer/bulk_index_job"
require "sidekiq/indexer/delete_job"
```

could become 


```
GovukIndexPageTrafficJob
GovukIndexPopularityJob
GovukIndexPublishingEventJob
IndexerAmendJob

etc.
```

If we wanted to adhere to the pattern

I don't feel massively strongly about it, but just thought it'd be worth giving the counter argument. Let me know what you think and i'm happy to go either way.

Reply:
As I understand this would be the only repo that'd be using a lib directory for them (as it's a Sinatra app) whereas the Rails apps all have the aspect that for `app/sidekiq` the first two directories aren't expected to be part of the Ruby namespace.

But to be honest my suggestion is somewhat trying to be pragmatic of trying to balance one thing that is a bit weird (now) with lack of ability to properly follow sidekiq conventions (future because of lib directory) that I think there's no way it won't end up a bit janky.

Reply:
cool makes sense we'd follow a different pattern for Sinatra apps. I'll make those changes now